code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreStart.java

```diff
@@ -66,7 +66,12 @@ public void doWithLockNoResult() {
             Map<String, Object> driver_opts = DataAccessor.fieldMap(v, VolumeConstants.FIELD_VOLUME_DRIVER_OPTS);
             if (driver_opts.containsKey(VolumeConstants.EC2_AZ)) {
                 String zone = driver_opts.get(VolumeConstants.EC2_AZ).toString();
-                labels.put(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL, String.format("%s=%s", VolumeConstants.HOST_ZONE_LABEL_KEY, zone));
+                String label = String.format("%s=%s", VolumeConstants.HOST_ZONE_LABEL_KEY, zone);
+                if (labels.containsKey(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL)) {
```

Comment:
we still need to check if the object is null, labels is a map<String,Object>. I can do the check in one line.